akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -203,15 +211,20 @@ private[remote] abstract class InboundCompression[T >: Null](
 
   /* ==== COMPRESSION ==== */
 
+  /** Override and specialize if needed, for default compression logic delegate to 3-param overload */
+  def decompress(incomingTableVersion: Int, idx: Int): OptionVal[T]
+
   /**
    * Decompress given identifier into its original representation.
    * Passed in tableIds must only ever be in not-decreasing order (as old tables are dropped),
    * tableIds must not have gaps. If an "old" tableId is received the value will fail to be decompressed.
    *
    * @throws UnknownCompressedIdException if given id is not known, this may indicate a bug – such situation should not happen.
    */
-  // not tailrec because we allow special casing in sub-class, however recursion is always at most 1 level deep
-  def decompress(incomingTableVersion: Int, idx: Int): OptionVal[T] = {
+  @tailrec final def decompressInternal(incomingTableVersion: Int, idx: Int, attemptCounter: Int): OptionVal[T] = {
```

Comment:
saw it a few times where it ended up looping very deep and blowing up, with stackoverflow, throwing this exception instead now in case it happens again